### PR TITLE
05561 wiki 2

### PR DESCRIPTION
**Pull request by cody-littley**
First draft of a platform wiki. For rendered link, see https://github.com/hashgraph/hedera-services/blob/05561-wiki-2/platform-sdk/docs/platformWiki.md

### CODE REVIEW

File: platform-sdk/docs/platformWiki.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+This code is maintained by the "Platform Hashgraph" team.
+
+- Components
+    - Gossip
+        - Sync gossip algorithm
+        - Out of order gossip algorithm
+    - Hashgraph
+    -  State management
+    - State snapshots
```
Comment (poulok) [inline platform-sdk/docs/platformWiki.md:21]: indentation under State Management got lost

File: platform-sdk/docs/platformWiki.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    - Hashing
+    - State Signing
+    - ISS Detection
+    - Reconnect
+    - Transaction Handling
+    - BLS
+- Event Flow
```
Comment (poulok) [inline platform-sdk/docs/platformWiki.md:28]: Please add a place holder for `Event Intake` under `Event Flow`

File: platform-sdk/docs/platformWiki.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    - Pre-consensus event stream
+    - Post-consensus event stream
+    - Threading Diagram
+- State Proofs
```
Comment (poulok) [inline platform-sdk/docs/platformWiki.md:32]: Let's not include this just yet. It's not currently being worked

File: platform-sdk/docs/platformWiki.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+## Base
+
+This code is maintained by the "Platform Base" team.
+
+- [Configuration](./base/configuration/configuration.md)
+- Metrics
+    - Prometheus
+- Logging
+- Thread Management
+- Notification Engine
```
Comment (poulok) [inline platform-sdk/docs/platformWiki.md:43]: Is this really base? Right now it's in the `ApplicationCommunicationComponent`. Which also should be listed above under `Components`

**Comment by hendrikebbers** [comment]
I really like the idea to get our documentation structured in a better way. But I do not know if the doc folder with custom named MD files is the best idea. 

In general I would favor to use the GitHub Wiki features that let you easily define a minimal wiki (structure) for your repo. A good example can be found [here](https://github.com/Netflix/Hystrix/wiki) and a general overview can be found [here](https://github.com/MyHoneyBadger/awesome-github-wiki). Using this feature gives us support like custom header, footer and menu structure (all defined by markdown). 

Using the GitHub wiki has still some problems. You can for example not easily switch between different versions of the wiki. You will always see the latest version. For such functionality we would need to setup a system like https://antora.org or https://squidfunk.github.io/mkdocs-material/. I do not see that work for now.

Another point is the new repo structure. I already thought about that some days ago. We need to find a way to have platform and services documentation life together. Especially when the 2 become 1 gradle project you do not want to search in different doc folders that are totally separated. Topics like the gradle setup, Java format, etc. are global for the complete repo and we need to take care of that.

File: platform-sdk/docs/platformWiki.md (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/docs/platformWiki.md:21]: fixed

File: platform-sdk/docs/platformWiki.md (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/docs/platformWiki.md:28]: added

File: platform-sdk/docs/platformWiki.md (hunk 4 above)
Comment (cody-littley) [reply platform-sdk/docs/platformWiki.md:43]: Moved to components

**Comment by alittley** [comment]
@hendrikebbers Am I understanding correctly, that the documents included in the wiki are update outside the normal git flow? This don't seem desirable, since that means documentation can only be updated post-merge into develop. This decreased convenience doesn't seem worth the minor benefit of having a header / footer.

**Comment by david-bakin-sl** [comment]
(A possibility _for the future_† would be to move these pages _into_ the repo (the wiki is itself a git repo so you can make a patchset) and then publish them via github pages.  Then they'd be in the normal flow, always current on the github pages url, yet older versions available in the usual git/github way (especially since github will render markdown files in the repo nicely).)

----
†No need to do it now: start with this nice documentation in the wiki then it'll be a reasonably easy task to move it later to whatever publishing mechanism you'd like.  Since the wiki is itself a git repo ...

File: platform-sdk/docs/platformWiki.md (hunk 1 above)
Comment (poulok) [reply platform-sdk/docs/platformWiki.md:31]: I'm still not seeing the additional indentation showing items like `State snapshots` show as a sub bullet of `State management`

File: platform-sdk/docs/platformWiki.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    - Hashgraph
+    - State management
+    - State snapshots
+    - Hashing
+    - State Signing
+    - ISS Detection
+    - Reconnect
+    - Transaction Handling
+    - BLS
+    - Notification Engine
```
Comment (poulok) [inline platform-sdk/docs/platformWiki.md:38]: Change this to `Application Communication`

**Review by lpetrovic05 — APPROVED** [review]

**Review by alittley — APPROVED** [review]

**Review by hendrikebbers — DISMISSED** [review]

File: platform-sdk/docs/platformWiki.md (hunk 5 above)
Comment (cody-littley) [reply platform-sdk/docs/platformWiki.md]: done

File: platform-sdk/docs/platformWiki.md (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/docs/platformWiki.md]: Added indentation.

**Review by netopyr — APPROVED** [review]

**Review by poulok — APPROVED** [review]

**Pull request merged**